setup.cfg

```diff
@@ -31,3 +31,9 @@ filterwarnings =
 
 markers =
     expensive: mark a test as expensive (deselect with '-m "not expensive"')
+
+# yapf is a Python source formatter
+[yapf]
```

Comment:
Thinking about yapf again, I remember now that we already have `autopep8` configured to do the appropriate auto-indent via the [flake8] and [pep8] entries in `setup.cfg`.

It's prob better to keep only one of these two tools in our repo since they fulfill the same autoformatting function. I'm down for replacing `autopep8` with `yapf`. What do you think, @AdamGleave ?

I like the "one true formatting" philosophy of yapf because I had a great experience using `go fmt`. Don't think that we will likely end up piping the entire codebase through `yapf`, but would enjoy trying it.

Reply:
If we get rid of `autopep8` in setup.py, the [pep8] section of `setup.cfg` becomes superfluous and should be removed.

Reply:
What's our intended use case for these tools? I generally think we should either enforce usage of a particular tool by running it as part of CI, or not have it at all.

`autopep8` is a lot less opinionated than `yapf`. Running `autopep8` on our code, I think we should probably make all the changes it suggests (mostly there are some 4-space indents that should be changed), and plausibly this should be part of the CI checks. 

By contrast, `yapf` would make a lot of changes I'm not sure I agree with. I'm OK sucking it up as uniformity of style is more important, but if we apply `yapf` in a piecemeal basis we don't get that either.

So I'd say: bite the bullet and run `yapf` on the entire codebase, or stick with `autopep8`, and in either case they should be part of CI.

Reply:
(I agree if we have a setup.cfg entry we should probably have it be part of `dev`, though.)

Reply:
I just put this in so that I could use yapf for new code that I write. Personally I find yapf's "opinions" slightly annoying, but it integrates nicely with my editor, so I've used it over the rest of the formatters. For existing code, I usually just fix pep8 errors by hand, so I wasn't planning to yapf the rest of the code base or integrate it into the CI checks. Is it still fine to keep that entry in `setup.cfg`?